test/scale/scale_tester.go

```diff
@@ -0,0 +1,292 @@
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      statefulSet.Statefulset.Name,
+			Namespace: statefulSet.Namespace.Name,
+		},
+		Spec: scale.ScaleSpec{
+			Replicas: int32(0),
+		},
+	}
+	_, _ = client.AppsV1().StatefulSets(statefulSet.Namespace.Name).UpdateScale(context.TODO(), statefulSet.Statefulset.Name, newScale, metav1.UpdateOptions{})
+
+	numberOfRemainingPods := totalNumberOfPods
+	tickerCount = 0
+	stopLoop = false
+	go func() {
+		time.Sleep(11 * time.Minute)
+		tickerStop <- true
+	}()
+	for {
+		select {
+		case <-tickerStop:
+			stopLoop = true
+
+		case <-ticker.C:
+			tickerCount = tickerCount + 1
+			numberOfRunningPods = podutil.CountAllPodsWithMatchingLabel(client, namespace, map[string]string{"app": "azuredisk-volume-tester"}, string(v1.PodRunning))
+			if numberOfRemainingPods <= 0 {
+				stopLoop = true
+			}
+
+			if tickerCount == 1 {
+				oneMinPodsDeleted = totalNumberOfPods - numberOfRemainingPods
+			} else if tickerCount == 2 {
+				twoMinPodsDeleted = totalNumberOfPods - numberOfRemainingPods
+			} else if tickerCount == 3 {
+				threeMinPodsDeleted = totalNumberOfPods - numberOfRemainingPods
+			} else if tickerCount == 5 {
+				fiveMinPodsDeleted = totalNumberOfPods - numberOfRemainingPods
+			} else if tickerCount == 10 {
+				tenMinPodsDeleted = totalNumberOfPods - numberOfRemainingPods
+			}
+		}
+
+		if stopLoop {
+			break
+		}
+	}
+
+	scaleInPodsCompleted := time.Since(startScaleIn).Minutes()
+
+	for {
+		numberOfRemainingVolumeAtts := volutil.CountAllVolumeAttachments(client)
+		if numberOfRemainingVolumeAtts <= 0 || time.Since(startScaleIn) > time.Duration(*scaleTestTimeout)*time.Minute {
+			break
+		}
+	}
+	scaleInCompleted := time.Since(startScaleIn).Minutes()
+
+	klog.Infof("Scaling out test completed in %f minutes.", scaleOutCompleted)
+	klog.Infof("1 min: %d pods are ready", oneMinPodsRunning)
+	klog.Infof("2 min: %d pods are ready", twoMinPodsRunning)
+	klog.Infof("3 min: %d pods are ready", threeMinPodsRunning)
+	klog.Infof("5 min: %d pods are ready", fiveMinPodsRunning)
+	klog.Infof("10 min: %d pods are ready", tenMinPodsRunning)
+	klog.Infof("Total number of pods in Running state: %d", numberOfRunningPods)
+
+	klog.Infof("Scaling in test completed in %f minutes.", scaleInPodsCompleted)
+	klog.Infof("1 min: %d pods are deleted", oneMinPodsDeleted)
+	klog.Infof("2 min: %d pods are deleted", twoMinPodsDeleted)
+	klog.Infof("3 min: %d pods are deleted", threeMinPodsDeleted)
+	klog.Infof("5 min: %d pods are deleted", fiveMinPodsDeleted)
+	klog.Infof("10 min: %d pods are deleted", tenMinPodsDeleted)
+	klog.Infof("Total number of remaining pods: %d", numberOfRemainingPods)
+
+	klog.Infof("Scaling in detach completed in %f minutes.", scaleInCompleted)
+}
+
+func (t *PodSchedulingOnFailoverScaleTest) Run(client clientset.Interface, namespace *v1.Namespace, schedulerName string) {
+	// Get the list of available nodes for scheduling the pod
+	nodes := nodeutil.ListNodeNames(client)
+	if len(nodes) < 3 {
```

Comment:
This check is unnecessary, it is not checking anything valuable for the scale test. We need a way bigger number than 3 here and that number will change depending on the vm type and number of disks.